EnginePrimeExportJob: Use mixxx::audio::FramePos internally

Holzhaus opened the pull request.

Holzhaus committed:
EnginePrimeExportJob: Use mixxx::audio::FramePos internally

uklotzde commented on src/library/export/engineprimeexportjob.cpp:
```diff
@@ -177,10 +167,10 @@ void exportMetadata(djinterop::database* pDatabase,
-    snapshot.default_main_cue = cueSampleOffset;
-    snapshot.adjusted_main_cue = cueSampleOffset;
+    mixxx::audio::FramePos cuePlayPos = pTrack->getMainCuePosition();
+    // FIXME: What if the cuePlayPos is invalid?
```
@mr-smidge Please verify and handle these cases after we merged this PR.

mr-smidge replied:
Under what circumstances can the main cue position be invalid?  If not set?

I suppose the simplest solution is just to set to zero/start-of-track in the export.

uklotzde commented on src/library/export/engineprimeexportjob.cpp:
```diff
@@ -204,14 +193,13 @@ void exportMetadata(djinterop::database* pDatabase,
         const auto lastBeatPlayPos =
-                beats->findPrevBeat(mixxx::audio::FramePos::fromEngineSamplePos(
-                        sampleOffsetToPlayPos(sampleCount)));
+                beats->findPrevBeat(mixxx::audio::FramePos(frameCount));
```
`mixxx::audio::kStartFramePose + frameCount`? Doesn't matter but would be the correct calculation.

Holzhaus committed:
EnginePrimeExportJob: Use sum of start frame pos and frame count

Requested here: https://github.com/mixxxdj/mixxx/pull/4077#discussion_r666116858

uklotzde reviewed (approved):
LGTM
